src/main/scala/TestTaskHomework.scala

```diff
@@ -0,0 +1,215 @@
+import java.io.{File, FileWriter}
+import java.nio.file.{Files, Paths}
+import scala.io.Source
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try, Using}
+
+object TestTaskHomework {
+
+  type ErrorMessage = String
+  val errorPrefix      = "Error: "
+  val cellErrorMessage = "#Incorrect message"
+  final case class RawSpreadsheet(size: Size, table: Seq[String])
+  final case class ProcessedSpreadsheet(size: Size, cells: Seq[Cell])
+  final case class Size(rowCnt: Int, colCnt: Int)
+  final case class Cell(result: String)
+
+  def validateReadPath(readPath: Option[String]): Either[ErrorMessage, String] = {
+    readPath match {
+      case None => Left(errorPrefix + "Input file doesn't exist")
+      case Some(path) =>
+        if (Files.exists(Paths.get(path))) Right(path) else Left(errorPrefix + "Input file doesn't exist")
+    }
+
+  }
```

Comment:
You can make it even shorter

```suggestion
  def validateReadPath(readPath: Option[String]): Either[ErrorMessage, String] = readPath match {
      case None => Left(errorPrefix + "Input file doesn't exist")
      case Some(path) =>
        if (Files.exists(Paths.get(path))) Right(path) else Left(errorPrefix + "Input file doesn't exist")
  }
```